src/avatar/__tests__/avatar.test.js

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+
+import React from 'react';
+import {mount} from 'enzyme';
+
+import {Avatar} from '../index';
+
+// manually trigger src load error. jsdom will not fire a load event
+// https://github.com/jsdom/jsdom/issues/1816#issuecomment-310106280
+function triggerLoadError(wrapper) {
+  wrapper.instance().handleError();
+  wrapper.update();
+}
+
+describe('Avatar', () => {
+  it('applies expected accessibility attributes to img element', () => {
+    const name = 'user name';
+    const wrapper = mount(<Avatar name={name} src="valid-img-src.png" />);
+
+    const image = wrapper.find('img');
+    expect(image.getDOMNode().getAttribute('alt')).toBe(name);
+  });
+
+  it('does not render img element if image fails to load', () => {
+    const wrapper = mount(
+      <Avatar name="user name" src="invalid-img-src.png" />,
+    );
+
+    expect(wrapper.find('img')).toHaveLength(1);
+    triggerLoadError(wrapper);
+    expect(wrapper.find('img')).toHaveLength(0);
+  });
+
+  it('applies expected accessibility attributes if image fails to load', () => {
+    const name = 'user name';
+    const wrapper = mount(<Avatar name={name} src="invalid-img-src.png" />);
+    const rootNode = wrapper.childAt(0).getDOMNode();
+
+    expect(rootNode.getAttribute('aria-label')).toBeNull();
```

Comment:
these `expects` refer to another logic, when image doesn't fail to load. I would split this test into two, so these `expects` do not confuse with the main `expects` of this test. And we can combine them under one reasonable `describe` to combine common `arrange` part

Reply:
good idea